test/integration/consensus/consensus_sunny_day.cpp

```diff
@@ -36,17 +39,32 @@ auto mk_local_peer(uint64_t num) {
 class FixedCryptoProvider : public MockYacCryptoProvider {
  public:
   explicit FixedCryptoProvider(const std::string &public_key) {
-    pubkey.fill(0);
-    std::copy(public_key.begin(), public_key.end(), pubkey.begin());
+    auto tmp =
+        shared_model::crypto::DefaultCryptoAlgorithmType::generateKeypair()
+            .publicKey();
+    std::string key(tmp.blob().size(), 0);
+    std::copy(public_key.begin(), public_key.end(), key.begin());
```

Comment:
Why this so complicated it is just strings?

Reply:
Because I need the key to be a specific size